api/src/tests/unit/utils/Paginator.test.ts

```diff
@@ -0,0 +1,55 @@
+import { ModelStatic } from "sequelize";
+import Paginator from "../../../utils/Paginator";
+import { testPaginationAssets } from "../../helpers/testData";
+import { MockedLogger } from "../../helpers/mockLogger";
+
+jest.mock("../../../utils/Logger.ts", (): MockedLogger => ({
+  info: jest.fn(),
+  error: jest.fn(),
+}));
+
+const MockedModel: jest.Mocked<ModelStatic<any>> =
+  jest.createMockFromModule("sequelize");
+const mockFindAndCountAll = jest.fn();
+MockedModel.findAndCountAll = mockFindAndCountAll;
+
+describe("Paginator Unit Tests", () => {
+  let paginator: Paginator<any>;
+
+  beforeEach(() => {
+    paginator = new Paginator(MockedModel);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("returns paginated test data with 3 total records", async () => {
+
+    mockFindAndCountAll.mockResolvedValueOnce({ rows: testPaginationAssets.data, count: testPaginationAssets.totalRecords });
+
+    const page = 1;
+    const pageSize = 10;
+    const result = await paginator.paginate(page, pageSize);
+
+    expect(result.lastPage).toBe(1);
+    expect(result.totalRecords).toBe(3);
```

Comment:
W: Make dynamic to be `testPaginationAssets.length`, to be resilient to new test assets being added.

Reply:
Commit: 02dbeb7a0fa4367798ed906384e2bba6522c1763